src/libcore/iter/mod.rs

```diff
@@ -520,6 +520,35 @@ impl<I> Iterator for Cycle<I> where I: Clone + Iterator {
 #[unstable(feature = "fused", issue = "35602")]
 impl<I> FusedIterator for Cycle<I> where I: Clone + Iterator {}
 
+/// An iterator that skips n elements every iteration.
+///
+/// This `struct` is created by the [`every_nth`] method on [`Iterator`]. See
+/// its documentation for more.
+///
+/// [`every_nth`]: trait.Iterator.html#method.every_nth
+/// [`Iterator`]: trait.Iterator.html
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+#[stable(feature = "rust1", since = "1.0.0")]
+#[derive(Clone, Debug)]
+pub struct EveryNth<I> {
+    iter: I,
+    step: usize,
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<I> Iterator for EveryNth<I> where I: Iterator {
+    type Item = I::Item;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        let elt = self.iter.next();
+        if self.step > 0 {
+            self.iter.nth(self.step - 1);
```

Comment:
I can think of a way to avoid fusing. But that would involve consuming an element when `every_nth` is called, which might be against the "laziness" principle of iterator combinators.